isis/appdata/import/cassis.lbl.tpl

```diff
@@ -20,94 +20,93 @@ Object = IsisCube
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window3StartLine             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window3_Start_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window3EndLine               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window3_End_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window4Binning               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Binning_window_4}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window4StartSample           = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window4_Start_Col}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window4EndSample             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window4_End_Col}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window4StartLine             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window4_Start_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window4EndLine               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window4_End_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window5Binning               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Binning_window_5}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window5StartSample           = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window5_Start_Col}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window5EndSample             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window5_End_Col}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window5StartLine             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window5_Start_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window5EndLine               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window5_End_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window6Binning               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Binning_window_6}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window6StartSample           = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window6_Start_Col}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window6EndSample             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window6_End_Col}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window6StartLine             = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window6_Start_Row}}{%endif%}
+    {%if exists("Product_Observational.CaSSIS_Header")%}Window6EndLine               = {{Product_Observational.CaSSIS_Header.PEHK_HEADER.attrib_Window6_End_Row}}{%endif%}
   End_Group
 
   Group = BandBin
```

Comment:
Unsure how to handle the rest of the groups. There is a trn for updating the BandBin but its inconsistent with the labels I have seen. Not sure how Kernel and AlphaCube groups are generated.

Reply:
The Kernels is a good question. Right now, I believe it will always be as follows for any CaSSIS image, so I don't think it needs to be replaced in the template. 

```
  Group = Kernels
    NaifFrameCode = -143400
  End_Group
```

For the `AlphaCube` group generation, check out line 457 - 462 in `isis/src/tgo/apps/tgocassis2isis/main.cpp`. You may need to write a callback function like Tim did in his PR to get this part working.

For the `BandBin` group, the NaifIkCode comes from line 417-428 in `isis/src/tgo/apps/tgocassis2isis/main.cpp`

Reply:
My confusion on the TgoCassisMosaicBandBin.trn and TgoCassisBandBin.trn files is that in the label I see this 
```` 
  Group = BandBin
    FilterName = BLU
    Center     = 485 <nm>
    Width      = 165 <nm>
    NaifIkCode = -143424
  End_Group
````
But in the .trn it says it should be translated to 497.4 if BLU for center and 134.3 if BLU for width. So I am not sure why these are different.